### PR TITLE
Pulp docker registry

### PR DESCRIPTION
**Pull request by climebil**
Deploy docker registry container

### CODE REVIEW

File: modules/opencontrail_ci/manifests/pulp_repo.pp [hunk 1]
```diff
@@ -66,18 +71,44 @@
...
+  selinux::port { 'crane':
+    argument => '-m',
+    context  => http_port_t,
+    protocol => tcp,
+    port     => 5001,
   }
 
-  firewall { '100 accept all to 80 - repos over http ':
+  docker::image { 'registry': }
+  docker::run { 'registry':
```
Comment (wurbanski) [inline modules/opencontrail_ci/manifests/pulp_repo.pp]: I think this needs a volume for persistence.

File: modules/opencontrail_ci/manifests/pulp_repo.pp [hunk 2]
```diff
@@ -66,18 +71,44 @@
...
+
+  selinux::port { 'crane':
+    argument => '-m',
+    context  => http_port_t,
+    protocol => tcp,
+    port     => 5001,
   }
 
-  firewall { '100 accept all to 80 - repos over http ':
+  docker::image { 'registry': }
```
Comment (wurbanski) [inline modules/opencontrail_ci/manifests/pulp_repo.pp]: Is this needed? Docker usually pulls image if it does not exist locally. Does docker::image keep the latest image downloaded?

**Review by diabelko — APPROVED** [review]

File: modules/opencontrail_ci/manifests/pulp_repo.pp (hunk 2 above)
Comment (climebil) [reply modules/opencontrail_ci/manifests/pulp_repo.pp]: docker::run can pull the image if it doesn't exist, however it has no means to control tag/version. Explicit containter tag has been added.

File: modules/opencontrail_ci/manifests/pulp_repo.pp (hunk 1 above)
Comment (climebil) [reply modules/opencontrail_ci/manifests/pulp_repo.pp]: Persistent volume (bind mount) for registry data has been added.

File: modules/opencontrail_ci/manifests/pulp_repo.pp [hunk 3]
```diff
@@ -81,11 +81,30 @@
...
 
-  docker::image { 'registry': }
+  file { '/docker-registry':
+    ensure => directory,
+    owner  => root,
+    group  => root,
+    mode   => '0700',
+  }
+
+  file { '/docker-registry/data':
```
Comment (wurbanski) [inline modules/opencontrail_ci/manifests/pulp_repo.pp:91]: Doesn't this require the one above?

**Pull request merged**